pkg/cvo/availableupdates.go

```diff
@@ -95,12 +108,24 @@ func (u *availableUpdates) NeedsUpdate(original *configv1.ClusterVersion) *confi
 
 // setAvailableUpdates updates the currently calculated version of updates.
 func (optr *Operator) setAvailableUpdates(u *availableUpdates) {
+	success := false
 	if u != nil {
-		u.At = time.Now()
+		u.LastAttempt = time.Now()
+		if u.Condition.Type == configv1.RetrievedUpdates {
+			success = u.Condition.Status == configv1.ConditionTrue
+		} else {
+			klog.Warningf("Unrecognized condition %s=%s (%s: %s): cannot judge update retrieval success", u.Condition.Type, u.Condition.Status, u.Condition.Reason, u.Condition.Message)
+		}
 	}
 
 	optr.statusLock.Lock()
 	defer optr.statusLock.Unlock()
+	if u != nil && (optr.availableUpdates == nil ||
+		optr.availableUpdates.Upstream != u.Upstream ||
+		optr.availableUpdates.Channel != u.Channel ||
+		success) {
+		u.LastSyncOrConfigChange = u.LastAttempt
+	}
```

Comment:
Is it okay to set LastSyncOrConfigChange even when we had "Unrecognized condition"? Depends on the semantics of LastSyncOrConfigChange but it can't be "the CVO is succesfully retrieving updates" unless Status is ConditionTrue.

Reply:
> Is it okay to set `LastSyncOrConfigChange` even when we had "Unrecognized condition"?

When you change `Upstream` or `Channel`, yes, because that's the `OrConfigChange` portion of this.  The `LastSync` portion is switching on `success`, which I have `true` on:

```go
if u.Condition.Type == configv1.RetrievedUpdates {
  success = u.Condition.Status == configv1.ConditionTrue
} ...
```

so it will be `false` in all "Unrecognized condition" cases, right?

Reply:
But 'if u != nil && optr.availableUpdates == nil' LastSyncOrConfigChange gets regardless of Condition.Status. Should it just be based on 'success'?

Reply:
> `if u != nil && optr.availableUpdates == nil`.

This is "we're setting this for the first time".  Might mean we're resetting this value whenever we restart a CVO, which isn't great (and would be fixed by having the CVO populate `optr.availableUpdates` from an in-cluster ClusterVersion on start), but othewise I don't see a problem with this.

Reply:
I was thinking of it failing on the first time, init, and it setting Status to False but if it fails on the first time it leaves it nil so, yeah, should be fine.